assets/javascripts/discourse/controllers/event-rsvp.js.es6

```diff
@@ -14,23 +15,23 @@ export default Ember.Controller.extend({
     const type = this.get('type');
     const topic = this.get('model.topic');
 
-    let usernames = topic.get(`event_${type}`);
+    let user_ids = topic.get(`event_${type}`);
 
-    if (!usernames || !usernames.length) return;
+    if (!user_ids || !user_ids.length) return;
 
     let userList = [];
 
-    usernames.forEach((username, index) => {
-      User.findByUsername(username).then((user) => {
-        userList.push(user);
-
-        if (userList.length == usernames.length) {
-          this.setProperties({
-            userList,
-            loadingList: false
-          })
-        }
-      })
+    ajax('/calendar-events/rsvp/get_users', {
+      data:{
+        user_ids: user_ids
+      },
+      type: "POST"
+    }).then((response) => {
+      let userList = response.users;
+        this.setProperties({
+          userList,
+          loadingList: false
+        });
     });
```

Comment:
@fzngagan We need to add a ``catch`` statement here to handle errors. Errors should be displayed with ``flash`` functionality in the ``modal-functionality`` mixin.